spec.html

```diff
@@ -5638,6 +5638,23 @@ <h1>IsStringPrefix ( _p_, _q_ )</h1>
       </emu-note>
     </emu-clause>
 
+    <emu-clause id="sec-isstringvalidunicode" aoid="IsStringValidUnicode">
+      <h1>Static Semantics: IsStringValidUnicode ( _string_ )</h1>
+      <p>
+        The abstract operation IsStringValidUnicode takes an argument _string_ ( a String ). It interprets _string_ as a sequence of UTF-16 encoded code points, as described in <emu-xref href="#sec-ecmascript-language-types-string-type"></emu-xref>, and returns if it is a complete valid UTF-16 sequence. It performs the following steps when called:
+      </p>
+      <emu-alg>
+        1. Let _strLen_ be the number of code units in string.
+        1. Let _k_ be 0.
+        1. Repeat,
+          1. If _k_ equals _strLen_, return.
```

Comment:
should this return true or false?

Reply:
ah, that should be breaking out of the loop, it should make the whole algorithm return `true`

Reply:
In that case, I'd expect `Repeat, while _k_ does not equal _strLen_,` or similar.

Reply:
should be fixed

Reply:
thanks, this works, but i think it'd be clearer with a `return true` at the end, and the condition as part of the `repeat`

Reply:
redone

Reply:
Thanks! now this line isn't needed anymore tho:
```suggestion
```